File: Javascript/0051-0100/100. Same Tree.js
// https://leetcode.com/problems/same-tree/description/

var isSameTree = function(p, q) {
    if(p == null && q == null)
        return true; //reached end (or) empty tree

    if(p == null || q == null)
        return false; //any one of the node becomes null

    //check current node value
    if(p.val != q.val)
        return false; //value not-matching

    //check for left and right sides
    return isSameTree(p.left,q.left) && isSameTree(p.right,q.right);
};

File: Python/0051-0100/100. Same Tree.py
# https://leetcode.com/problems/same-tree/description/

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p == None and q == None:
            return True #reached end (or) empty tree

        if p == None or q == None:
            return False; # any one of the node becomes None

        # check current node value
        if p.val != q.val:
            return False # value not-matching

        # check for left and right sides
        return self.isSameTree(p.left,q.left) and self.isSameTree(p.right,q.right)
